hack/lib/kube-versions.txt

```diff
@@ -12,9 +12,10 @@
 # Whenever a new version is added to this file, or when a version number
 # is edited in this file, please run hack/update.sh.
 #
-1.30.3
-1.29.7
-1.28.12
+1.31.0
+1.30.4
+1.29.8
+1.28.13
```

Comment:
Should we remove 1.24 since we are adding 1.31? Oldest one out when new one in?

Reply:
If so, also `rm -rf` that generated directory.

Reply:
I meant to discuss that. As things are there aren't any pressing reasons to remove them, and I know in the past we've removed older versions that were causing maintenance burden.

Reply:
I think last time we added support for a new Kube version we decided that we would drop the oldest generated code directory each time, but leave the old version of Kube in CI for running the integration tests.

The cost of generating code for too many kube versions is that it slows down calls to `update.sh` both for manual invocations and for the CI verify job. It's not a big deal, but we need to drop the old versions from the generated directory at some point (can't keep adding forever), so last time we made up a policy and documented it in the comments of this file.

Reply:
Next push will remove generated code for K8s 1.24